pkg/apis/velero/v1/backup_storage_location.go

```diff
@@ -66,6 +66,10 @@ type BackupStorageLocationSpec struct {
 	Config map[string]string `json:"config"`
 
 	StorageType `json:",inline"`
+
+	// ReadOnly is whether the backup storage location is in read-only mode. If true,
+	// backups cannot be created or deleted in this location.
+	ReadOnly bool `json:"readOnly"`
```

Comment:
Alternately, this could be a `Mode` field, with values of "ReadWrite", "ReadOnly", etc.

Reply:
I think `Mode` is more flexible, in case we add any other states. I don't know what they would be at the moment, though. Maybe something for replication, such as `ReplicateOnly`?

Reply:
Yeah, I agree on the flexibility but also couldn't think of possible other states, but probably makes sense to go that way regardless.

Reply:
Oh, lol: https://github.com/heptio/velero/blob/master/pkg/apis/velero/v1/backup_storage_location.go#L83

Guess we defined this back when we added the BSL type, just never implemented it.

Reply:
Hmm, this ended up as a `status` field, though, not sure that makes sense.